src/DefaultBuilder/src/WebApplicationBuilder.cs

```diff
@@ -34,31 +37,30 @@ internal WebApplicationBuilder(Assembly? callingAssembly, string[]? args = null)
             Environment = _environment = new WebHostEnvironment(callingAssembly);
 
             Configuration.SetBasePath(_environment.ContentRootPath);
-            Services.AddSingleton(Environment);
 
             // Run methods to configure both generic and web host defaults early to populate config from appsettings.json
             // environment variables (both DOTNET_ and ASPNETCORE_ prefixed) and other possible default sources to prepopulate
             // the correct defaults.
-            var bootstrapBuilder = new BootstrapHostBuilder(Configuration, _environment);
-            bootstrapBuilder.ConfigureDefaults(args);
-            bootstrapBuilder.ConfigureWebHostDefaults(configure: _ => { });
-            bootstrapBuilder.RunConfigurationCallbacks();
-
-            Logging = new LoggingBuilder(Services);
-            WebHost = _deferredWebHostBuilder = new ConfigureWebHostBuilder(Configuration, _environment, Services);
-            Host = _deferredHostBuilder = new ConfigureHostBuilder(Configuration, _environment, Services);
+            _bootstrapHostBuilder = new BootstrapHostBuilder(Configuration, _environment, Services);
+            _bootstrapHostBuilder.ConfigureDefaults(args);
+            _bootstrapHostBuilder.ConfigureWebHostDefaults(webHostBuilder =>
+            {
+                // Runs inline.
+                webHostBuilder.Configure(ConfigureApplication);
```

Comment:
```suggestion
                webHostBuilder.Configure(ConfigureApplication);

                // We need to override the application name since the call to Configure will set it to
                // be the calling assembly's name.
                webHostBuilder.UseSetting(WebHostDefaults.ApplicationKey, _environment.ApplicationName);
```

Reply:
@halter73 this should fix the issue.